src/presentation/pages/Classification.tsx

```diff
@@ -55,16 +57,17 @@ const Classification = () => {
     setDraggingSubmissionId(null)
     const { active, over } = event
 
-    let grade: AssessmentGrade | null = null
-    let rank: AssessmentRank | null = null
+    let grade: AssessmentGradeOfFrontend | null = null
+    let rank: AssessmentRankOfFrontend | null = null
 
     if (over && active.id !== over?.id) {
       const newItems = report.items.map((item) => {
         if (item.student.numId === active.id) {
           if (over.id !== 'has-not-grade') {
             const [newGrade, newRank] = (over.id as string).split(':')
-            grade = Number(newGrade) as AssessmentGrade
-            rank = newRank as AssessmentRank
+            const finalNewRank = newRank === '' ? undefined : newRank
+            grade = Number(newGrade) as AssessmentGradeOfFrontend
+            rank = finalNewRank as AssessmentRankOfFrontend
```

Comment:
```suggestion
            newRank = newRank === '' ? undefined : newRank
            grade = Number(newGrade) as AssessmentGradeOfFrontend
            rank = newRank as AssessmentRankOfFrontend
```
finalNewRankとは更新後のrankを指すと思うのですが、finalとnewの反対の単語を同時に使うと流石に他の方から見たときに分かりづらいので、newRankに再代入すればよいと思います。newRankのconst [newGrade,newRank]からlet [newGrade,newRank]にすると再代入できます。